compiler/circle-resizer/tests/ShapeParser.test.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ShapeParser.h"
+
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <string>
+#include <tuple>
+#include <vector>
+
+using namespace circle_resizer;
+
+class ParseShapeTestFixture : public ::testing::TestWithParam<std::tuple<std::string, Shapes>>
+{
+};
+
+TEST_P(ParseShapeTestFixture, successful_parsing)
+{
+  const auto [input_shape_str, expected_shapes] = GetParam();
+  Shapes result_shapes;
+  EXPECT_NO_THROW(result_shapes = parse_shapes(input_shape_str));
+  ASSERT_EQ(result_shapes, expected_shapes);
+}
+
+INSTANTIATE_TEST_SUITE_P(
+  ParseShapeTest, ParseShapeTestFixture,
+  ::testing::Values(
```

Comment:
Some models may have a scalar input (rank = 0, size = 1). How scalar is expressed in the user input of `circle-resizer`?

Reply:
Good point! I've refactored `Dim` to represent a scalar using `std::optional`. For a moment I've thought about `Shape{}` but it can be misleading with an not initialized shape

Reply:
I've imagined that an empty `std::vector<Dim>` would be a scalar. I didn't see your design, but using a separate class (as you mentioned) would be clearer. :)

Reply:
@jinevening thank you for the quick feedback ;) So I'll use the current design with a separate class.